src/dispatcher.rs

```diff
@@ -0,0 +1,117 @@
+use crate::commands::Commands;
+use serenity::{model::prelude::*, prelude::*, utils::parse_username, Client};
+use std::collections::HashMap;
+
+pub(crate) struct MessageStore;
+
+impl TypeMapKey for MessageStore {
+    type Value = HashMap<String, (Message, ChannelId)>;
+}
+
+impl MessageStore {
+    pub(crate) fn init(client: &mut Client) {
+        let mut data = client.data.write();
+        data.insert::<Self>(HashMap::new());
+    }
+
+    pub(crate) fn save(cx: &Context, name: String, msg: (Message, ChannelId)) {
+        let mut data = cx.data.write();
+        let store = data
+            .get_mut::<Self>()
+            .expect("Unable to access message store.  ");
```

Comment:
Inconsistency with `expect` message endings. Some end in a period and two spaces, and others done.

Reply:
Nixed